ui/pages/confirmations/hooks/useCurrentConfirmation.ts

```diff
@@ -48,27 +51,28 @@ const useCurrentConfirmation = () => {
         ({ id: confirmId }) => confirmId === paramsTransactionId,
       );
     }
+
     if (!pendingConfirmation) {
       if (!latestPendingConfirmation) {
         setCurrentConfirmation(undefined);
         return;
       }
       pendingConfirmation = latestPendingConfirmation;
     }
+
     if (pendingConfirmation.id !== currentConfirmation?.id) {
       const unconfirmedTransaction =
         unconfirmedTransactions[pendingConfirmation.id];
       if (!unconfirmedTransactions) {
         setCurrentConfirmation(undefined);
         return;
       }
-      if (
-        pendingConfirmation.type !== ApprovalType.PersonalSign &&
-        pendingConfirmation.type !== ApprovalType.EthSignTypedData
-      ) {
+
+      if (!ConfirmationRedesignTypes.includes(pendingConfirmation.type)) {
```

Comment:
this causes a TS error 
<img width="1045" alt="Screenshot 2024-04-24 at 3 37 16 PM" src="https://github.com/MetaMask/metamask-extension/assets/20778143/65aa3c0b-ecc7-4de2-b570-781ddf8e93cd">

I'm working on updating the expected ApprovalRequest type through our core packages
https://github.com/MetaMask/core/pull/4209

Reply:
Updated the code to allow both ApprovalType and string types in the meantime [096a49a](https://github.com/MetaMask/metamask-extension/pull/24182/commits/096a49a3887f7c8f650bd15ea7db4307ef0782c5)